Build setup tweaked - having es, cjs and umd files now

Andarist opened the pull request:
fixes #58

Changes:
- eslint update
- some new tools got used like `rollup`, `prettier` etc
- using `src` as module entry point as no es6+ features are used (except the export ofc)
- `lib` is built from `src` as commonjs file (main entry point)
- umd builds are also built into `dist`

Due to nature of changes this probably should get released as new major and I thought I will also include 2 breaking changes along the way (I can ofc revert any of them if you want to):
- removed deprecated functionalities
- dropped support for ie8 (removed `bind` "polyfill")

taylorhakes commented on src/index.js:
```diff
@@ -0,0 +1,216 @@
+  }
+}
+
+export default function Promise(fn) {
```
Could you move this export to the bottom of the file? I prefer defining all exports at the bottom.

```
export default Promise;
```

Andarist replied:
moved

Andarist commented:
> If you could add back the bind polyfill, I will merge this PR.

Brought this back.

I've also changed node's version on travis to the latest, version 4 was needed for some dev deps but also npm4 is needed for publication (I've added `prepublish` which tests that to prevent publications on older npms which wouldn't run `prepare` script) so I've just made it `latest`. I don't think testing this on 0.12 was giving any extra guarantees anyway so I consider this bump not an issue - let me know if you think otherwise.

taylorhakes commented:
Ok, I would prefer to test on 0.12 because it doesn't support Promise, but I think it's fine. 

Thanks again for all this work. I will look to create a new version in the next couple days

Andarist commented:
No problem, those were really simple adjustments needed to make the whole thing work
